### PR TITLE
Move TaskUtils.makeId to AbstractTask

### PR DESCRIPTION
**Pull request by drcrallen**

### CODE REVIEW

File: indexing-service/src/main/java/io/druid/indexing/common/task/TaskUtils.java [hunk 1]
```diff
@@ -26,8 +26,7 @@
 {
   public static String makeId(String id, final String typeName, String dataSource, Interval interval)
```
Comment (himanshug) [inline indexing-service/src/main/java/io/druid/indexing/common/task/TaskUtils.java]: I think we should just remove TaskUtils class as it only contains one method which is really just calling another method now. All the usage of this method are inside Task classes which can directly use AbstractTask.joinId.

Comment (nishantmonu51) [inline indexing-service/src/main/java/io/druid/indexing/common/task/TaskUtils.java]: or probably move the IDJoiner and joinId method to TaskUtil.

Comment (drcrallen) [inline indexing-service/src/main/java/io/druid/indexing/common/task/TaskUtils.java]: I thought about removing it, but it does provide some standard structure over the general joiner. Meaning that it is expected that an ID will be in the `type_datasource_start_end_timestamp` format, while still retaining the arbitrary capacity of AbstractTask.joinId

Comment (himanshug) [inline indexing-service/src/main/java/io/druid/indexing/common/task/TaskUtils.java]: Hmmm, In that case, I guess we can move this method to AbstractTask itself and remove this class.

Comment (drcrallen) [inline indexing-service/src/main/java/io/druid/indexing/common/task/TaskUtils.java]: moved

**Comment by himanshug** [comment]
+1

**Pull request merged**